jdaviz/core/template_mixin.py

```diff
@@ -184,6 +184,16 @@ def session(self):
     def data_collection(self):
         return self._app.session.data_collection
 
+    def _clear_cache(self, *attrs):
```

Comment:
In principle, it is nice but the API can be potentially confusing. Why the choice to use `*attrs` instead of something more explicit?

Reply:
What if someone accidentally pass in some value that is intrinsic to Python built-ins and breaks everything when cleared?

Reply:
They are the names of attributes (technically cached properties, but they act like attributes while they are cached) on the class whose cache are getting cleared.  Passing in any invalid string (something that isn't actually cached) is just ignored because of the if-statement.

Do you have any suggestions?  This is what we use elsewhere (currently in the base plugin class), but we can change it there as well (or I suppose put it in its own mixin or monkeypatch it onto `VuetifyTemplate` so that it only exists once in our codebase).

Reply:
> What if someone accidentally pass in some value that is intrinsic to Python built-ins and breaks everything when cleared deleted?

I guess we could have some safety measures that inspects the class and does checking, but this is meant for internal use.... so don't try to clear something that is a python-internal and only use it for things that are cached properties!

Reply:
Is there a way for `_clear_cache` to check to see which attrs are actually our own caching keys and so ok to delete? Surely we know in advance what we're supposed to cache?

Reply:
we could either require `self._cached_properties` to be set and require any entry to be in there, or possibly some inspection of the methods (within the loop, check to make sure that there is a method with the same name and perhaps inspect to see if it is decorated as a cached property).  I'm just not sure if it's worth the overhead for the checks and I just simplified this a bit more so `self._cached_properties` is only required if you want to call without explicitly passing arguments (in which case it just defaults to clearing everything stored in `self._cached_properties`).

> Surely we know in advance what we're supposed to cache?

If that is the argument, then surely we wouldn't maliciously try to clear something that wasn't cached.